### ISSUE TITLE
Not working with dotenv

### ISSUE THREAD

**Issue by hyperh** [visionmedia/debug#446]
I can't seem to get `debug` working with `dotenv`. It works fine when I set the env var from outside of .env like so:

package.json
```
"dev": "NODE_ENV=development DEBUG=*,apicache nodemon",
```

```
npm run dev
```

server.js
```
import 'babel-polyfill';
import makeDebug from 'debug';
import dotenv from 'dotenv';

const env = dotenv.config();
console.log(process.env.DEBUG); // outputs *,apicache correctly
const debug = makeDebug('brainfm:server/index');
debug('.env', env); // Doesn't show anything
```

.env
```
DEBUG =*,apicache
```

**Comment by thebigredgeek** [comment]
@hyperh  , this is likely an issue with `dotenv`  We aren't doing anything magical with how we intake environment variables.  Can you file an issue there and paste a link to the issue here?

**Comment by FredyC** [comment]
Isn't this more likely an issue of timing when env is initialized by dotenv and read by debug? In your example you are importing debug and then later configuring dotenv. I am not sure where this is actually handled on debug side, I don't actually see it in a source code.

Generally I recommend adding npm script like `"start": "node -r dotenv/config -r babel-polyfill server.js"`. It is generally better approach like instead of hardwiring these things into code.

**Comment by thebigredgeek** [comment]
We get the environment variable within the load function in `node.js` and `browser.js`, [example](https://github.com/visionmedia/debug/blob/master/src/browser.js#L158).  We immediately call `enable`, passing the result of `load()` [here](https://github.com/visionmedia/debug/blob/master/src/browser.js#L168)... so all of the environment variables are uptaken when debug is first imported.  You should be able to call `debug.enable` and pass whatever pattern you want at runtime if you are dynamically intaking environment variables as shown [here](https://github.com/visionmedia/debug/blob/master/src/debug.js#L138), but make sure to do this before any debug instances are created as it won't work for instances that are created after you call `enable`.

**Comment by thebigredgeek** [comment]
fixed with 2.6.5

**Comment by iwaduarte** [comment]
@thebigredgeek  It took me a few years to find this comment. Most people just ignore it and replace it eventually if they use the dotenv approach. I guess this should be added to the documentation. Working with dotenv should be done as below:

``` 
  [esm]
import  dotenv from "dotenv";
dotenv.config();

import debugFactory from 'debug';
const { DEBUG } = process.env;
debugFactory.enable(DEBUG)

const debug = debugFactory('whatever');
debug("Hello dotenv");

```
